### PR TITLE
fix: fake behavior with special replacement patterns

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes the issue described here: https://github.com/faker-js/faker/pull/503#issuecomment-1079744428

### CODE REVIEW

File: src/fake.ts [hunk 1]
```diff
@@ -43,32 +43,25 @@ export class Fake {
...
-    let res = '';
-
     // if incoming str parameter is not provided, return error message
     if (typeof str !== 'string' || str.length === 0) {
       throw new Error('string parameter is required!');
     }
 
     // find first matching {{ and }}
-    const start = str.search('{{');
+    const start = str.search('{{[a-z]');
```
Comment (Shinigami92) [inline src/fake.ts:52]: This looks like a regex :thinking: 
Can `search` can take a real regex? Then we should use `str.search(/\{\{[a-z]/)`

Comment (ST-DDT) [reply src/fake.ts]: Fixed

File: test/fake.spec.ts [hunk 2]
```diff
@@ -50,5 +50,39 @@ describe('fake', () => {
...
+      expect(faker.fake('}}hello{{random.alpha}}')).toMatch(/^}}hello[a-z]$/);
+    });
+
+    it('should be able to handle connected brackets', () => {
+      expect(faker.fake('{{{random.alpha}}}')).toMatch(/^{[a-z]}$/);
+    });
+
+    it('should be able to handle empty brackets', () => {
+      expect(faker.fake('{{}}')).toBe('{{}}');
+    });
```
Comment (ST-DDT) [inline test/fake.spec.ts:78]: Please check whether these tests match your expectation regarding the `fake()` method.

I'm especially unsure about `{{}}`.

File: src/fake.ts [hunk 3]
```diff
@@ -43,32 +43,25 @@ export class Fake {
...
     // if incoming str parameter is not provided, return error message
     if (typeof str !== 'string' || str.length === 0) {
       throw new Error('string parameter is required!');
     }
 
     // find first matching {{ and }}
-    const start = str.search('{{');
-    const end = str.search('}}');
+    const start = str.search(/{{[a-z]/);
+    const end = str.substring(start).search(/}}/) + start;
```
Comment (xDivisionByZerox) [inline src/fake.ts:53]: `end` can be determined by using `indexOf` instead of search since a RegEx is not required.

Comment (xDivisionByZerox) [reply src/fake.ts:53]: This would also drop the required `substring` call upfront since `indexOf` accepts a start position.

Comment (ST-DDT) [reply src/fake.ts]: Fixed

**Review by Shinigami92 — APPROVED** [review]

File: test/fake.spec.ts (hunk 2 above)
Comment (xDivisionByZerox) [reply test/fake.spec.ts:78]: I don't know to be honest. But these tests and the expected results seem reasonable to me.

**Review by import-brain — APPROVED** [review]

**Pull request merged**